sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -2100,3 +2100,53 @@ object AsOfJoin {
     }
   }
 }
+
+
+/**
+ * A logical plan for summary.
+ */
+case class Summary(
```

Comment:
I think this can be the proto plan for Spark Connect, and the server can translate it into DataFrame transformations, sharing the code with `df.summary`

Reply:
do you mean this way? https://github.com/apache/spark/pull/38318/files#diff-ecb87acd2b45a267cefe538c1cb2ad6de6a1094800757f757e64d2cb98cba7bdR99-R110

Reply:
Ideally it should better have dedicated SQL plan for each operation in DataFrame. I guess doing the rewrite and conversion in Analyzer is correct approach by right .. and Spark Connect just has the proto plan for the dedicated logical plan.

Reply:
That said, over time, Spark connector planner should only construct logical plans, but not DataFrame transformations?

Reply:
Maybe we can support them but I suppose it's better to keep them as exceptions. If most of them are logical plans, it should be easier for other people to follow the protocol and implement too. That's actually what I talked w/ @grundprinzip too (correct me if I am wrong)

Reply:
then a new plan LGTM. But we should probably name it `UnresolvedSummary`

Reply:
since `df.summary` had been reimplemented with dataframe operations https://github.com/apache/spark/commit/6a0713a141fa98d83029d8388508cbbc40fd554e, I guess the only difference is that:

with a new plan, the analysis can be saved before the job execution; but if there is something wrong in analysis, the exception will not be thrown.

Reply:
>  but if there is something wrong in analysis, the exception will not be thrown.

@zhengruifeng can you elaborate this a bit more? Is this a regression in terms of bad cases (e.g. users expect to exceptions but now it is gone)?

Reply:
if we use `df.summary` here, an analysis exception will be thrown immediately; if we use a new unresolved plan, I guess the exception will not be thrown before job execution.

Reply:
I see, by using a logical plan, we delay the analysis until the spark connect planner constructs the full plan.

Reply:
> But we should probably name it UnresolvedSummary

Reply:
they are not a function in my opinion, but more like a dataframe operation. e.g. we won't call `df.filter` a DataFrame function.

Reply:
ok let me rename it